src/main/scala/cromiam/webservice/CromIamApiService.scala

```diff
@@ -108,41 +108,51 @@ trait CromIamApiService extends Directives with SprayJsonSupport with DefaultJso
 
   val allRoutes: Route = workflowRoutes ~ engineRoutes
 
-  private def handleRequest(f: (Authorization, HttpRequest) => ToResponseMarshallable): Route = {
-    headerValuePF { case a: Authorization => a } { authorization =>
-      toStrictEntity(300.millis) {
-        extractRequest { req =>
-          complete {
-            f.apply(authorization, req)
+  private def handleRequestWithAuthn(directive: Directive0)(f: (Authorization, HttpRequest) => ToResponseMarshallable): Route = {
+    directive {
+      headerValuePF { case a: Authorization => a } { authorization =>
+        toStrictEntity(Timeout) {
+          extractRequest { req =>
+            complete {
+              f.apply(authorization, req)
```

Comment:
If I'm following correctly, I would concur that this is not worth the hassle of further DRY-ing.

Only three lines could be refactored into say a `completeWithTimeout`, as the call to `headerValuePF` breaks up the "duplicated" code. Perhaps when more duplications happen this can be refactored then.

Reply:
The part that always gets me is the part that's really changing is the innermost bit and it has a different signature. So your choices are to shoehorn the same signature, which seems yucky, or duplicate. And thus for cases like this I wind up not bothering.